include/llvm/ADT/ilist.h

```diff
@@ -248,20 +248,20 @@ void operator+(ilist_iterator<T>,int) = delete;
 
 // operator!=/operator== - Allow mixed comparisons without dereferencing
 // the iterator, which could very likely be pointing to end().
-template<typename T>
-bool operator!=(const T* LHS, const ilist_iterator<const T> &RHS) {
+template<typename T, typename U>
```

Comment:
Can you wrap your changes here in `// HLSL Change` comments. We try to mark out the core parts of LLVM that we've modified.

Reply:
Will do. Out of curiosity, what's the purpose of the `// HLSL Change` comments? I assumed they were to aid in updating DXC to newer versions of LLVM, but that seems unlikely now. Do these comments help you with the merge into mainline LLVM?

Reply:
Done

Reply:
There are two reasons these comments help. One is that we do cherry-pick back selective fixes from newer LLVM versions, so marking out the comments allow us to identify where we changed something in a merge. The other is that I frequently diff DXC's code against LLVM 3.7 to help narrow down the causes of bugs. More often than not, the bugs in some parts of DXC are caused by the way changes were introduced into the LLVM-inherited code, so having the markers to denote those areas is extremely usefu.

Reply:
Good to know. Thanks for the additional info!